src/www/ui/api/Helper/DbHelper.php

```diff
@@ -286,50 +286,92 @@ public function getUsers($id = null)
+        END AS job_status
+      FROM job j
+      LEFT JOIN jobqueue jq ON j.job_pk=jq.jq_job_fk
+      GROUP BY j.job_pk, j.job_queued, j.job_name, j.job_upload_fk, j.job_user_fk, j.job_group_fk
+    )";
+  }
+
   /**
    * @brief Get the recent jobs.
    *
    * If a limit is passed, the results are trimmed. If an ID is passed, the
    * information for the given id is only retrieved.
    *
    * @param integer $id       Set to get information of only given job id
+   * @param integer $status   Set to get information of only jobs with given status
+   * @param string  $sort     Set to sort the results asc or desc
    * @param integer $limit    Set to limit the result length
    * @param integer $page     Page number required
    * @param integer $uploadId Upload ID to be filtered
+   * @param integer $userId      Set to get information of only given user's ID
    * @return array[] List of jobs at first index and total number of pages at
    *         second.
    */
-  public function getJobs($id = null, $limit = 0, $page = 1, $uploadId = null)
+  public function getJobs($id = null, $status = null, $sort = "ASC", $limit = 0, $page = 1, $uploadId = null, $userId = null)
   {
-    $jobSQL = "SELECT job_pk, job_queued, job_name, job_upload_fk," .
-      " job_user_fk, job_group_fk FROM job";
-    $totalJobSql = "SELECT count(*) AS cnt FROM job";
+    $jobsWithStatusCteSQL = $this->getJobStatusCteSQLStatement();
+    $jobSQL = "$jobsWithStatusCteSQL SELECT * FROM job_with_status_cte";
+    $totalJobSql = "$jobsWithStatusCteSQL SELECT count(*) AS cnt FROM job_with_status_cte";
 
-    $filter = "";
     $pagination = "";
-
     $params = [];
-    $statement = __METHOD__ . ".getJobs";
+    $filter = [];
+    $statement = $userId !== null ? __METHOD__ . ".getUserJobs" : __METHOD__ . ".getJobs";
     $countStatement = __METHOD__ . ".getJobCount";
-    if ($id == null) {
-      if ($uploadId !== null) {
-        $params[] = $uploadId;
-        $filter = "WHERE job_upload_fk = $" . count($params);
-        $statement .= ".withUploadFilter";
-        $countStatement .= ".withUploadFilter";
-      }
-    } else {
+
+    if ($id !== null) {
       $params[] = $id;
-      $filter = "WHERE job_pk = $" . count($params);
+      $filter[] = "job_pk = $" . count($params);
       $statement .= ".withJobFilter";
       $countStatement .= ".withJobFilter";
+    } elseif ($uploadId !== null) {
+      $params[] = $uploadId;
+      $filter[] = "job_upload_fk = $" . count($params);
+      $statement .= ".withUploadFilter";
+      $countStatement .= ".withUploadFilter";
     }
 
-    $result = $this->dbManager->getSingleRow("$totalJobSql $filter;", $params,
-      $countStatement);
+    // if userId was given, add it to the where filter
+    if ($userId !== null) {
+      $params[] = $userId;
+      $filter[] = "job_user_fk = $" . count($params);
+    }
 
+    // if status was given, add it to the where filter
+    if ($status !== null) {
+      $params[] = $status;
```

Comment:
We should check the provided value for $status is also the expected value. Instead of directly appending it into SQL statement.

Something like what is done for $sort in line 373